source/winAPI/displayTracking.py

```diff
@@ -33,51 +36,45 @@ class OrientationState:
 _orientationState = OrientationState()
 
 
-def _updateOrientationState(height: int, width: int) -> bool:
+def _getNewOrientationState(
+		previousState: OrientationState,
+		height: int,
+		width: int,
+) -> Optional[Orientation]:
 	"""
-	@returns: True if there has been an orientation state change.
+	@returns: Orientation if there has been an orientation state change, otherwise None
 	"""
-	# Resolution detection comes from an article found at https://msdn.microsoft.com/en-us/library/ms812142.aspx.
-	heightAndWidthUnchanged = _orientationState.height == height and _orientationState.width == width
-	orientationChanged = False
-	if width > height:
-		# The new orientation is landscape
-		if (
-			# Orientation has changed
-			_orientationState.style != Orientation.LANDSCAPE
-			# If the height and width are the same, it's a screen flip
-			# otherwise, it may be a change of display (e.g. monitor disconnected).
-			or heightAndWidthUnchanged
-		):
-			_orientationState.style = Orientation.LANDSCAPE
-			orientationChanged = True
-	else:
-		# The new orientation is portrait
-		if (
-			# Orientation has changed
-			_orientationState.style != Orientation.PORTRAIT
-			# If the height and width are the same, it's a screen flip
-			# otherwise, it may be a change of display (e.g. monitor disconnected).
-			or heightAndWidthUnchanged
-		):
-			_orientationState.style = Orientation.PORTRAIT
-			orientationChanged = True
-
-	_orientationState.height = height
-	_orientationState.width = width
-	return orientationChanged
+	heightAndWidthUnchanged = previousState.height == height and previousState.width == width
+	newOrientation = Orientation.LANDSCAPE if width > height else Orientation.PORTRAIT
+	if (
+		# Orientation has changed
+		previousState.style != newOrientation
+		# If the height and width are the same, it's a screen flip
+		# and the orientation state has changed.
+		# Otherwise, it may be a change of display (e.g. monitor disconnected).
+		or heightAndWidthUnchanged
+	):
+		return newOrientation
+	return None
 
 
 def reportScreenOrientationChange(heightWidth: int) -> None:
 	"""
 	Reports the screen orientation only if the screen orientation has changed.
 	"""
+	# Resolution detection comes from an article found at
+	# https://msdn.microsoft.com/en-us/library/ms812142.aspx.
 	height = winUser.HIWORD(heightWidth)
 	width = winUser.LOWORD(heightWidth)
-	if _updateOrientationState(height, width):
+	newState = _getNewOrientationState(_orientationState, height, width)

```

Comment:
Given this returns the style part only, maybe this name is misleading.

Reply:
This has been renamed to _getNewOrientationStyle